super-cool-game/Assets/Scripts/Player/PlayerMovement.cs

```diff
@@ -1,72 +1,52 @@
-﻿using System.Collections;
-using System.Collections.Generic;
-using UnityEngine;
+﻿using UnityEngine;
 
 public class PlayerMovement : MonoBehaviour {
-    /* Handles the movement of the player.
-     * This script is always attached to the player's character game object,
-     * while PlayerInput is attached only when the character's owner is the local player.
-     */
-    
-    // These ones can be edited in the inspector
-    public float speed = 3.0f;
-    public float maxSpeed = 9.0f;
-    public float jumpHeight = 6.0f;
-    public float checkGroundDistance = 1.0f;
-    public LayerMask groundLayer;
+    /* Translates commands from PlayerController into movement */
 
     private Rigidbody2D rb;
-    private float moveX = 0;
+    private BoxCollider2D boxCollider;
 
-    // Use this for initialization
+    private float friction = 0.4f;
+    private float velX = 0;
+    
     void Start() {
-        // Get Rigidbody from the GameObject
-        rb = gameObject.GetComponent<Rigidbody2D>();
-    }
-
-    // Fixed update is for Physics-based update
-    void FixedUpdate () {
-        ComputeMovement();
+        rb = GetComponent<Rigidbody2D>();
+        boxCollider = GetComponent<BoxCollider2D>();
     }
-
-    void ComputeMovement () {
-        // Resultant of movement
-        rb.velocity = new Vector2(moveX, rb.velocity.y);
-
-        // Add force if speed is below max threshold
-        if(moveX * rb.velocity.x < maxSpeed) {
-            rb.AddForce(new Vector2(moveX * speed, 0.0f), ForceMode2D.Impulse);
-        }
-        // Cap speed
-        if(Mathf.Abs(rb.velocity.x) > maxSpeed) {
-            rb.velocity = new Vector2(Mathf.Sign(rb.velocity.x) * maxSpeed, rb.velocity.y);
-        }
-
-        // TODO: Flip if moveX < 0
+    
+    void FixedUpdate() {
+        rb.velocity = new Vector2(ApplyFriction(velX), rb.velocity.y);
     }
 
-    public void Jump () {
-        // if player hasnt already jumped
-        if(IsGrounded) {
-            rb.AddForce(new Vector2(0, jumpHeight), ForceMode2D.Impulse);
-        } 
+    public void Jump (float velY) {
+        if(IsGrounded)
+            rb.velocity = new Vector2(rb.velocity.x, velY);
     }
 
     // Moves the player on the X axis
-    public void Move (float axisX) {
-        moveX = axisX;
+    public void Move (float velX) {
+        this.velX = velX;
     }
 
     // Uses a raycast to check if player is grounded
     public bool IsGrounded {
         get {
-            Vector2 position = transform.position;
+            Vector2 origin = transform.position + Vector3.down * boxCollider.size.y;
```

Comment:
`Vector2.down` is equal to `Vector3.down`, but prefer the 2D one.

Reply:
transform.position is a Vector3, and you can't do operation on the two different types.
This adds them as Vector3 and then cast them onto a Vector2.